code/game/list.c

```diff
@@ -70,7 +70,7 @@ static int list_add_impl(list_p list, void* data, dataType_t type, size_t size,
     if (isPtr) {
         node->cont->data = malloc(size);
         if (node->cont->data == NULL) {
-            return 0;
+            goto on_error;
```

Comment:
It has its uses. It's to prevent having to duplicate cleanup code in multiple places. A lot of large C projects that I've encountered use it. It's not bad when used effectively.